### ISSUE TITLE
x.py ignores config.toml's debug-assertions-std for unit tests

### ISSUE THREAD

**Issue by ssomers** [rust-lang/rust#79481]
If you have in your config.toml file (or whichever file --config says), only this change to config.toml.example:

```toml
debug-assertions-std = true
```

debug assertions in src/liballoc or library/alloc used to be armed. Since #73374, debug assertions only go off if you also turn on `debug`. EDIT: since some point, even that doesn't help.

**Comment by jyn514** [comment]
@ssomers I'm still not able to replicate this. I ran `x test library/alloc --stage 0 --test-args debug_assert` with your test case added and the test failed appropriately:
```
Testing alloc stage0 (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
   Compiling getrandom v0.1.14
   Compiling ppv-lite86 v0.2.8
   Compiling rand_core v0.5.1
   Compiling rand_chacha v0.2.2
   Compiling rand_xorshift v0.2.0
   Compiling rand v0.7.3
   Compiling alloc v0.0.0 (/home/joshua/src/rust/rust/library/alloc)
    Finished release [optimized + debuginfo] target(s) in 1m 15s
     Running unittests (build/x86_64-unknown-linux-gnu/stage0-std/x86_64-unknown-linux-gnu/release/deps/alloc-cd5eecf891e8cfac)

running 1 test
F
failures:

---- collections::btree::node::tests::test_debug_assert stdout ----
thread 'collections::btree::node::tests::test_debug_assert' panicked at 'assertion failed: false', library/alloc/src/collections/btree/node/tests.rs:106:5
```

**Comment by ssomers** [comment]
Well, I can still replicate it… on the master at the time. I located the fix at auto merge #90042 (25 October).

And it's really flawless now. The debug assertions detonate if `debug` or `debug-assertions-std` is true, or both, and you don't need to clean anything before building to see it.
